src/frontend/src/utils/condition.rs

```diff
@@ -352,9 +353,111 @@ impl Condition {
                 }
             }
 
-            Ok(Some((non_overlap_scan_ranges, Condition::true_cond())))
+            Ok(Some((non_overlap_scan_ranges, false)))
         } else {
-            Ok(None)
+            let mut scan_ranges = vec![];
+            for (scan_ranges_chunk, _) in disjunctions_result {
+                if scan_ranges_chunk.is_empty() {
+                    // full scan range
+                    return Ok(None);
+                }
+
+                scan_ranges.extend(scan_ranges_chunk);
+            }
+            scan_ranges.sort_by(|a, b| a.eq_conds.len().cmp(&b.eq_conds.len()));
```

Comment:
Should we also sort lower_bound for each range here?🤔

e.g. `SELECT * FROM orders_count_by_user WHERE (user_id < 10) or (user_id > 30) or (user_id > 5 and user_id < 15);`
got following plan:
```
batch_plan: |-
    BatchExchange { order: [], dist: Single }
    └─BatchFilter { predicate: (((orders_count_by_user.user_id < 10:Int32) OR (orders_count_by_user.user_id > 30:Int32)) OR ((orders_count_by_user.user_id > 5:Int32) AND (orders_count_by_user.user_id < 15:Int32))) }
      └─BatchScan { table: orders_count_by_user, columns: [orders_count_by_user.user_id, orders_count_by_user.date, orders_count_by_user.orders_count], scan_ranges: [orders_count_by_user.user_id < Int64(10), orders_count_by_user.user_id > Int64(30), orders_count_by_user.user_id > Int64(5) AND orders_count_by_user.user_id < Int64(15)], distribution: UpstreamHashShard(orders_count_by_user.user_id, orders_count_by_user.date) }
```

I think what we expect is:
```
batch_plan: |-
    BatchExchange { order: [], dist: Single }
    └─BatchFilter { predicate: (((orders_count_by_user.user_id < 10:Int32) OR (orders_count_by_user.user_id > 30:Int32)) OR ((orders_count_by_user.user_id > 5:Int32) AND (orders_count_by_user.user_id < 15:Int32))) }
      └─BatchScan { table: orders_count_by_user, columns: [orders_count_by_user.user_id, orders_count_by_user.date, orders_count_by_user.orders_count], scan_ranges: [orders_count_by_user.user_id < Int64(15), orders_count_by_user.user_id > Int64(30)], distribution: UpstreamHashShard(orders_count_by_user.user_id, orders_count_by_user.date) }
```

Reply:
I try to reorder the predicate: `SELECT * FROM orders_count_by_user WHERE (user_id < 10) or  (user_id > 5 and user_id < 15) or (user_id > 30);`
```
  batch_plan: |-
    BatchExchange { order: [], dist: Single }
    └─BatchFilter { predicate: (((orders_count_by_user.user_id < 10:Int32) OR ((orders_count_by_user.user_id > 5:Int32) AND (orders_count_by_user.user_id < 15:Int32))) OR (orders_count_by_user.user_id > 30:Int32)) }
      └─BatchScan { table: orders_count_by_user, columns: [orders_count_by_user.user_id, orders_count_by_user.date, orders_count_by_user.orders_count], scan_ranges: [orders_count_by_user.user_id <= Int64(15), orders_count_by_user.user_id > Int64(30)], distribution: UpstreamHashShard(orders_count_by_user.user_id, orders_count_by_user.date) }
```

Why we get `orders_count_by_user.user_id <= Int64(15)` rather than `orders_count_by_user.user_id < Int64(15)` here?🤔